drivers/syslog/syslog_device.c

```diff
@@ -83,8 +79,7 @@ struct syslog_dev_s
   uint8_t      sl_state;    /* See enum syslog_dev_state */
   uint8_t      sl_oflags;   /* Saved open mode (for re-open) */
   uint16_t     sl_mode;     /* Saved open flags (for re-open) */
-  sem_t        sl_sem;      /* Enforces mutually exclusive access */
-  pid_t        sl_holder;   /* PID of the thread that holds the semaphore */
+  rmutex_t     sl_lock;     /* Enforces mutually exclusive access */
```

Comment:
Done in nxrmutex_unlock

Reply:
> I mean that current implementation of `nxmutex_` is lacking of `holder` information and allows releasing from the task that didn't acquired the mutex and here recursive mutex used only because it has the `holder` field, but `sl_lock` does not have recursive nature.

So you want to add hold to the normal mutex? since the lock is used in many place, it's better to save the holder only in debugging.

Reply:
good to have it for debug purposes in general. But let's keep it as is now.